client/rest/src/main/java/org/elasticsearch/client/RuntimeInfo.java

```diff
@@ -0,0 +1,145 @@
+        if (version != null) {
+            s.append(",hl=").append(version);
+        }
+
+        version= kotlinVersion();
+        if (version != null) {
+            s.append(",kt=").append(version);
+        }
+
+        version = scalaVersion();
+        if (version != null) {
+            s.append(",sc=").append(version);
+        }
+
+        version = clojureVersion();
+        if (version != null) {
+            s.append(",clj=").append(version);
+        }
+
+        version = groovyVersion();
+        if (version != null) {
+            s.append(",gy=").append(version);
+        }
+
+        return s.toString();
+    }
+
+    public static String HlrcKind() {
+        try {
+            Class.forName("org.elasticsearch.client.RestHighLevelClient");
+        } catch (Exception t) {
+            return null;
+        }
+        // 1 is the HLRC based on ES server request/response classes
+        // 2 will be the HLRC based on code generation from API specs
+        return "1";
+    }
+
+    public static String kotlinVersion() {
+        try {
+            //KotlinVersion.CURRENT.toString()
+            Class<?> clazz = Class.forName("kotlin.KotlinVersion");
+            Field field = clazz.getField("CURRENT");
+            String version = field.get(null).toString();
+            return stripPatchRevision(version);
+
+        } catch (Exception t) {
+            // ignore
+        }
+        return null;
+    }
+
+    public static String scalaVersion() {
+        try {
+            // scala.util.Properties.versionNumberString()
+            Class<?> clazz = Class.forName("scala.util.Properties");
+            Method m = clazz.getMethod("versionNumberString");
+            String version = (String) m.invoke(null);
+            return stripPatchRevision(version);
+
+        } catch (Exception t) {
+            // ignore
+        }
+        return null;
+    }
+
+    public static String clojureVersion() {
+        try {
+            // (clojure-version) which translates to
+            // clojure.core$clojure_version.invokeStatic()
+            Class<?> clazz = Class.forName("clojure.core$clojure_version");
+            Method m = clazz.getMethod("invokeStatic");
+            String version = (String) m.invoke(null);
+            return stripPatchRevision(version);
+
+        } catch (Exception t) {
+            // ignore
+        }
+        return null;
+    }
+
+    public static String groovyVersion() {
+        try {
+            // groovy.lang.GroovySystem.getVersion()
+            // There's also getShortVersion(), but only since Groovy 3.0.1
+            Class<?> clazz = Class.forName("groovy.lang.GroovySystem");
+            Method m = clazz.getMethod("getVersion");
+            String version = (String) m.invoke(null);
+            return stripPatchRevision(version);
+
+        } catch (Exception t) {
+            // ignore
+        }
+        return null;
+    }
+
+    static String stripPatchRevision(String version) {
```

Comment:
If patch numbers aren't long can we keep this info anyways?

Reply:
They can be long depending on the language, like `2.13.0-M5-6e0cba7` for some release of Scala. We could also compact pre-release versions but there's honestly no real value in the patch revision for this particular use case where we just want to know if a particular language is present, and in what major version (minor language versions are often not exactly minor). The patch number won't give us any actionable information unless we have a dedicated client for that language, in which case the full version will be reported in the first part of the header.